src/plots/cartesian/axes.js

```diff
@@ -1834,18 +1850,14 @@ function numFormat(v, ax, fmtoverride, hover) {
                 ((exponentFormat === 'SI' || exponentFormat === 'B') &&
                  (exponent > 12 || exponent < -15))) {
             v += 'e' + signedExponent;
-        }
-        else if(exponentFormat === 'E') {
+        } else if(exponentFormat === 'E' && base === 10) {
             v += 'E' + signedExponent;
-        }
-        else if(exponentFormat === 'power') {
-            v += '&times;10<sup>' + signedExponent + '</sup>';
-        }
-        else if(exponentFormat === 'B' && exponent === 9) {
+        } else if(exponentFormat === 'B' && exponent === 9 && base === 10) {
             v += 'B';
-        }
-        else if(exponentFormat === 'SI' || exponentFormat === 'B') {
+        } else if((exponentFormat === 'SI' || exponentFormat === 'B') && (base === 2 || base === 10)) {
```

Comment:
:cow2: maybe assign `var isBase10 = (base === 10);` instead of checking it in an if-clause multiple times.